zcash_primitives/src/consensus.rs

```diff
@@ -484,6 +497,94 @@ impl BranchId {
         // Sprout rules apply before any network upgrade
         BranchId::Sprout
     }
+
+    /// Returns the range of heights for the consensus epoch associated with this branch id.
+    ///
+    /// The resulting tuple implements the [`RangeBounds<BlockHeight>`] trait.
+    pub fn height_range<P: Parameters>(&self, params: &P) -> Option<impl RangeBounds<BlockHeight>> {
+        self.height_bounds(params).map(|(lower, upper)| {
+            (
+                Bound::Included(lower),
+                upper.map_or(Bound::Unbounded, Bound::Excluded),
+            )
+        })
+    }
+
+    /// Returns the range of heights for the consensus epoch associated with this branch id.
+    ///
+    /// The return type of this value is slightly more precise than [`Self::height_range`].
```

Comment:
Explain how it is more precise (I don't see how), and document what the various return cases mean:
```suggestion
    /// The return type of this value is slightly more precise than [`Self::height_range`]:
    /// - `Some((x, Some(y)))` means ...
    /// - `Some((x, None))` means ...
    /// - `None` means ...
```

Reply:
`Unbounded` is not a valid lower bound.

Reply:
And `Included` is not valid for the upper bound. So there are a bunch of invalid states that `height_range` can represent that we don't want visible. For the purpose of the tests, proptest doesn't actually work with `RangeBounds`; only the concrete `Range`, `RangeFrom` and `RangeTo` types.